workers/social/lib/social/models/group/permissionset.coffee

```diff
@@ -202,10 +208,32 @@ module.exports = class JPermissionSet extends Module
       permissions = (p.permission for p in advanced).join ', '
 
       JPermissionSet.checkPermission client, advanced, this, rest,
-        (err, hasPermission) ->
+        (err, hasPermission, permissionIndex) ->
           args = [client, rest..., callback]
           if err then callback err
           else if hasPermission
+            # client._allowedPermissionIndex
+            #
+            # if you write something like this;
+            #
+            #   foo: permit
+            #
+            #     advanced: [
+            #       { permission   : 'some permission', validateWith: Validators.own }
+            #       {
+            #         permission   : 'access related'
+            #         validateWith : accessValidator ACCESSLEVEL.READ
+            #       }
+            #       { permission   : 'modify object', superadmin: yes }
+            #     ]
+            #
+            #     success: (client, callback) ->
+            #       ...
+            #
+            # and this permission is granted because you were a superadmin then
+            # client._allowedPermissionIndex will become 2. But if you are owner of
+            # this instance this time it will become 0 because the first rule granted.
+            args[0]._allowedPermissionIndex = permissionIndex
```

Comment:
this may lead to unintended problems. generally we dont care about the order of permission but this adds importance now. 
If someone unknowingly passes the client around ( client is populated in bongo, but we can pass it from one function to another in models ) and if that client has `_allowedPermissionIndex` could cause issues. 
generally positional indexes are evil for me